modules/pool/invest/lib/useInvest.ts

```diff
@@ -10,17 +10,32 @@ import { usePool } from '~/modules/pool/lib/usePool';
 
 export function useInvest() {
     const { pool } = usePool();
-    const { selectedOptions, inputAmounts, zapEnabled } = useInvestState();
+    const { selectedOptions, inputAmounts, zapEnabled, setSelectedOption } = useInvestState();
     const { getUserBalanceForToken, userPoolTokenBalances } = usePoolUserTokenBalancesInWallet();
     const { priceForAmount } = useGetTokens();
 
-    const selectedInvestTokens: GqlPoolToken[] = pool.investConfig.options.map((option) =>
-        selectedOptions[`${option.poolTokenIndex}`]
-            ? option.tokenOptions.find(
-                  (tokenOption) => tokenOption.address === selectedOptions[`${option.poolTokenIndex}`],
-              )!
-            : option.tokenOptions[0],
-    );
+    const selectedInvestTokens: GqlPoolToken[] = pool.investConfig.options.map((option) => {
+        if (option.tokenOptions.length > 1) {
+            if (selectedOptions[`${option.poolTokenIndex}`]) {
+                return option.tokenOptions.find(
+                    (tokenOption) => tokenOption.address === selectedOptions[`${option.poolTokenIndex}`],
+                )!;
+            } else {
+                const firstOptionWithNonZeroBalance = option.tokenOptions.find(
+                    (option) => getUserBalanceForToken(option.address) !== '0.0',
+                );
+                // if there is more than 1 option and 1 of the options has a non-zero balance use that one
+                if (firstOptionWithNonZeroBalance) {
+                    setSelectedOption(option.poolTokenIndex, firstOptionWithNonZeroBalance.address);
```

Comment:
This is a dangerous action here, you're setting the value of a reactive var in a top level scope (ie: not triggered), that will cause this logic to immediately re-evaulate itself.